app.py

```diff
@@ -28,96 +30,122 @@ def init_db():
 
 ###CREATION AND DELETION FUNCTIONS
 
-#Function to create a note
-def create_note(): 
-    pass 
-
-#function to delete a note
-def delete_note():
-    pass
-
-#function to create a tag
-def create_tag(): 
-    pass
-
-#function to delete a tag
-def delete_tag():
-    pass 
-
-#function to add content
-#I do not know if we need a separate function for edit content. I can explain the reasoning in class 
-def add_content(): 
-    pass 
-
-
-###SEARCHING FUNCTIONS
-
-#This will be called by the frontend for two different functions
-    #1) Viewing Content 
-    #2) PDF Conversion 
-def return_content():
-    pass 
-
-#function to search by title 
-def search_by_title():
-    pass 
-
-#function to search by modified date
-def search_by_modified_date():
-    pass 
-
-#function to search by created date
-def search_by_created_date():
-    pass
-
-#function to search by tag
-def search_by_tag(): 
-    pass
-
-
-
-###LISTING FUNCTIONS
+#Used to create notes and create tags
+@app.route('/create', methods=['POST'])
+def create(): 
+    conn = sqlite3.connect("note.db")
+    cursor = conn.cursor()
+    payload = request.json
+    try: 
+        title = payload.get('title')
+    except KeyError:
+        abort(206, "Data Entry Requires A Title")
+    
+    try: 
+        content = payload.get('content')
+    except KeyError: 
+        content = ""
+    
+    try: 
+        tags = payload.get('tag')
+    except KeyError: 
+        tags = ""
+    
+    if tags != "":
+        for value in tags: 
+            cursor.execute("INSERT INTO tags VALUES (?,?)", (title, value))
+    
+    else: 
+        cursor.execute("INSERT INTO notes VALUES (?,?,?,?)", (title,content,date.today(),date.today()))
 
-def list_by_title():
+    conn.close()
+        
+     
+#used to delete notes and delete tags
+@app.route('/delete', methods=['POST'])
+def delete():
     conn = sqlite3.connect("note.db")
     cursor = conn.cursor()
-    cursor.execute(f"""
-                   SELECT notes.title
-                   FROM notes
-                   """)
+    payload = request.json
+    try: 
+        title = payload.get('title')
+    except KeyError:
+        abort(206, "Data Entry Requires A Title")
+    
+    try: 
+        tags = payload.get('tag')
```

Comment:
the caller should not be required to know the tag of the note they are deleting. instead we should handle that. i think delete should look like this:
1. check if a note by the title passed exists in the note table (do our own query on the db for that)
2. if it doesn't exist, raise some sort of error
3. if it does exist, delete all entries in the tags table that have the corresponding title
4. delete the entry in the notes table that has the corresponding title